hocuspocus-server/src/index.ts

```diff
@@ -19,15 +20,19 @@ const hocuspocusServer = Server.configure({
     if (authInfo === "unauthorized") {
       throw new Error("Not authorized!");
     }
-
-    data.connection.readOnly = authInfo.role === "COMMENT-ONLY";
+    
+    data.connectionConfig.readOnly = authInfo.role === "COMMENT-ONLY";
   },
 
   extensions: [
     new SQLite({
-      database: "db.sqlite",
+      database: ":memory:",
+    }),
+    // TODO we can actually just do the auth check in here, and not need the server to inject the mark or anything
```

Comment:
This might actually really simplify things, if we choose to do the authorization at the hocuspocus layer instead, it would allow clients to safely make modifications to the threads on their side. To make it safe though, we need to inspect what exactly was updated (i.e.  they can only modify their own comments).